ably/realtime_conn_test.go

```diff
@@ -0,0 +1,74 @@
+package ably_test
+
+import (
+	"reflect"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/ably/ably-go/ably"
+	"github.com/ably/ably-go/ably/testutil"
+)
+
+func record(t *testing.T, spy map[int]int, done func() bool) (chan<- ably.State, func()) {
```

Comment:
I would have probably called it `recordStateUntil` and renamed `spy` to be `statesCount`.